tests/scripts/thread-cert/Cert_5_2_01_REEDAttach.py

```diff
@@ -0,0 +1,161 @@
+#  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+#  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+#  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+#  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+#  POSSIBILITY OF SUCH DAMAGE.
+#
+
+import time
+import unittest
+
+import node
+import mle
+import config
+
+LEADER = 1
+DUT_ROUTER1 = 2
+REED1 = 3
+ED1 = 4
+
+class Cert_5_2_01_REEDAttach(unittest.TestCase):
+    def setUp(self):
+        self.nodes = {}
+        for i in range(1, 5):
+            self.nodes[i] = node.Node(i)
+
+        self.nodes[LEADER].set_panid(0xface)
+        self.nodes[LEADER].set_mode('rsdn')
+        self.nodes[LEADER].add_whitelist(self.nodes[DUT_ROUTER1].get_addr64())
+        self.nodes[LEADER].enable_whitelist()
+
+        self.nodes[DUT_ROUTER1].set_panid(0xface)
+        self.nodes[DUT_ROUTER1].set_mode('rsdn')
+        self.nodes[DUT_ROUTER1].add_whitelist(self.nodes[LEADER].get_addr64())
+        self.nodes[DUT_ROUTER1].add_whitelist(self.nodes[REED1].get_addr64())
+        self.nodes[DUT_ROUTER1].enable_whitelist()
+        self.nodes[DUT_ROUTER1].set_router_selection_jitter(1)
+
+        self.nodes[REED1].set_panid(0xface)
+        self.nodes[REED1].set_mode('rsdn')
+        self.nodes[REED1].add_whitelist(self.nodes[DUT_ROUTER1].get_addr64())
+        self.nodes[REED1].add_whitelist(self.nodes[ED1].get_addr64())
+        self.nodes[REED1].enable_whitelist()
+        self.nodes[REED1].set_router_selection_jitter(1)
+        self.nodes[REED1].set_router_upgrade_threshold(1)
+
+        self.nodes[ED1].set_panid(0xface)
+        self.nodes[ED1].set_mode('rsn')
+        self.nodes[ED1].add_whitelist(self.nodes[REED1].get_addr64())
+        self.nodes[ED1].enable_whitelist()
+        
+        self.sniffer = config.create_default_thread_sniffer()
+        self.sniffer.start()
+
+    def tearDown(self):
+        self.sniffer.stop()
+        del self.sniffer
+
+        for node in list(self.nodes.values()):
+            node.stop()
+        del self.nodes
+
+    def test(self):
+        self.nodes[LEADER].start()
+        self.nodes[LEADER].set_state('leader')
+        self.assertEqual(self.nodes[LEADER].get_state(), 'leader')
+
+        # 1 DUT_ROUTER1: Attach to LEADER
+        self.nodes[DUT_ROUTER1].start()
+        time.sleep(10)
+        self.assertEqual(self.nodes[DUT_ROUTER1].get_state(), 'router')
+
+        # DUT_ROUTER1: Verify MLE advertisements
+        router1_messages = self.sniffer.get_messages_sent_by(DUT_ROUTER1)
+        msg = router1_messages.next_mle_message(mle.CommandType.ADVERTISEMENT)
+        msg.assertSentWithHopLimit(255)
+        msg.assertSentToDestinationAddress('ff02::1')
+        msg.assertMleMessageContainsTlv(mle.SourceAddress)
+        msg.assertMleMessageContainsTlv(mle.LeaderData)
+        msg.assertMleMessageContainsTlv(mle.Route64)
```

Comment:
There is a method(check_mle_advertisement) in command.py.  You can reuse it here.

Reply:
Good idea. These commands have been encapsulated in command.py.